File: javascript/examples/latn-iast-demo.js
#!/usr/bin/env node
import {
    IastPlainEnglishOptions,
    PlainEnglishRomanizationProfile,
    TransliterationResult,
    toDevanagari,
    toGujarati,
    toPlainEnglish,
} from '../src/index.js';

const source = 'Kṛṣṇa ā́tman ḷa';
const devanagari = toDevanagari(source);
const gujarati = toGujarati(source);
const hunterian = toPlainEnglish(source, {
    options: new IastPlainEnglishOptions({
        profile: PlainEnglishRomanizationProfile.HUNTERIAN,
    }),
});

console.log(devanagari.rendered);
console.log(gujarati.rendered);
console.log(hunterian.rendered);

if (hunterian.restoreOriginal() !== source) {
    throw new Error('Exact source restoration failed.');
}

const serialized = JSON.stringify(hunterian.toJson());
const restoredEnvelope = TransliterationResult.fromJson(JSON.parse(serialized));
if (restoredEnvelope.restoreOriginal() !== source) {
    throw new Error('Serialized envelope restoration failed.');
}

File: dart/example/latn_iast_demo.dart
import 'dart:convert';

import 'package:lipimala/transliteration_result.dart';

void main() {
  const source = 'Kṛṣṇa ā́tman ḷa';

  final devanagari = source.toDevanagari();
  final gujarati = source.toGujarati();
  final hunterian = source.toPlainEnglish(
    options: const IastPlainEnglishOptions(
      profile: PlainEnglishRomanizationProfile.hunterian,
    ),
  );

  print(devanagari.rendered);
  print(gujarati.rendered);
  print(hunterian.rendered);

  // The exact original is available even from a deliberately lossy view.
  assert(hunterian.restoreOriginal() == source);

  final serialized = jsonEncode(hunterian.toJson());
  final restoredEnvelope = TransliterationResult.fromJson(
    Map<String, Object?>.from(jsonDecode(serialized) as Map),
  );
  assert(restoredEnvelope.restoreOriginal() == source);
}
